sageopt/relaxations/sage_polys.py

```diff
@@ -17,15 +17,56 @@
 from sageopt import coniclifts as cl
 from sageopt.symbolic.polynomials import Polynomial, PolyDomain
 from sageopt.symbolic.signomials import Signomial
+from sageopt.coniclifts.base import Expression
 from sageopt.relaxations import sage_sigs
 from sageopt.relaxations.sage_sigs import _check_kwargs
 from sageopt.relaxations import constraint_generators as con_gen
 from sageopt.relaxations import symbolic_correspondences as sym_corr
 
 
+def create_covers(s):
+    covers = {}
+
+    # If it is an expression, check to see the ScalarExpressions are constant
+    if isinstance(s.c, Expression):
+        for i in range(s.c.size):
+            # If s.c is positive, then do not need covers[i]
+            if s.c[i].is_constant() and s.c[i].value >= 0 and np.all(s.alpha[i, :] % 2 == 0):
+                pass
+            else:
+                covers[i] = np.full(shape=s.c.size, fill_value=True, dtype=bool)
+                covers[i][i] = False
+    # Otherwise it is a numeric ndarray
+    else:
+        for i in range(s.c.size):
+            # Check if positive constant
+            if s.c[i] >= 0 and np.all(s.alpha[i, :] % 2 == 0):
+                pass
+            else:
+                covers[i] = np.full(shape=s.c.size, fill_value=True, dtype=bool)
+                covers[i][i] = False
```

Comment:
Actually, you can skip this by replacing ``c = Expression(s.c)`` at the very beginning, and then accessing `c` instead of `s.c`. If `s.c` was already an Expression then this does nothing, otherwise it casts `s.c` to an Expression. This removes the need for the `isinstance` check.

Reply:
Right, that makes sense, the `isinstance` check was removed.